From e92c91b6b45d228164c960414edb5fb84b35b584 Mon Sep 17 00:00:00 2001
From: Matthias Klumpp <matthias@tenstral.net>
Date: Fri, 11 Mar 2022 19:09:38 +0100
Subject: [PATCH] Consider files with an mtime of 0 as well

Somehow the mtimes of the files are all at 0 in Flatpak, which trips up
imagej-launcher and makes it refuse to launch anything.
---
 src/main/c/file-funcs.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main/c/file-funcs.c b/src/main/c/file-funcs.c
index 56340a0..3cc4a2a 100644
--- a/src/main/c/file-funcs.c
+++ b/src/main/c/file-funcs.c
@@ -399,7 +399,7 @@ char *find_jar(const char *jars_directory, const char *prefix)
 			continue;
 		string_set_length(buffer, length);
 		string_append(buffer, name);
-		if (!stat(buffer->buffer, &st) && st.st_mtime > mtime) {
+		if (!stat(buffer->buffer, &st) && st.st_mtime >= mtime) {
 			free(result);
 			result = strdup(buffer->buffer);
 			mtime = st.st_mtime;
-- 
2.35.1

